### ISSUE TITLE
[BUG] Parse/Date has fatal error with php8 

### ISSUE THREAD

**Issue by poetfreak** [simplepie/simplepie#686]
> **IMPORTANT:** If you are posting code or output, please make sure you wrap it with [fenced code blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks/) so that it is readable.

#### Description
Tried to use the page from https://simplepie.org/wiki/setup/sample_page to test simplepie.
php 8 returns fatal errors about arrays.

#### Steps to Reproduce

using the second example, I changed 
```
$feed->set_feed_url
```
to
```
$feed->set_feed_url('https://www.usnews.com/rss/news');
```

it gets the title and description perfectly, then I get:

**_Fatal error: Uncaught TypeError: implode(): Argument #2 ($array) must be of type ?array,
string given in /home/a/public_html/simplepie-1.5/library/SimplePie/Parse/Date.php:544_**
Stack trace: #0 /home/a/public_html/simplepie-1.5/library/SimplePie/Parse/Date.php(544): implode()
#1 /home/a/public_html/simplepie-1.5/library/SimplePie/Parse/Date.php(577): SimplePie_Parse_Date->__construct()
#2 /home/a/public_html/simplepie-1.5/library/SimplePie/Registry.php(221): SimplePie_Parse_Date::get()
#3 /home/a/public_html/simplepie-1.5/library/SimplePie/Item.php(805): SimplePie_Registry->call()
#4 /home/a/public_html/simplepie-1.5/library/SimplePie.php(3203): SimplePie_Item->get_date()
#5 [internal function]: SimplePie::sort_items()
#6 /home/a/public_html/simplepie-1.5/library/SimplePie.php(3115): usort()
#7 /home/a/public_html/simplepie-1.5/demo/test1.php(32): SimplePie->get_items()
#8 {main} thrown in /home/a/public_html/simplepie-1.5/library/SimplePie/Parse/Date.php on line 544

#### Expected Behavior

I assume it would gracely warn and continue regardless of system date or feed date ?!?

#### Actual Behavior

Fatal death?

#### Possible Solutions

I guess I can try to see the date setup in php... but other then that this seems like a php8 issue since I believe array functions have changed and many warnings are now errors.

#### Environment
php 8.08 on Apache/2.4.38 (Debian)

**Comment by poetfreak** [comment]
![date](https://user-images.githubusercontent.com/8329437/124532592-7922dd80-ddd6-11eb-9a06-e3a28e1ee902.png)

**Comment by jrfnl** [comment]
@poetfreak What version are you using ? IIRC this issue was fixed in #617, which was included in Simplepie 1.5.3, so I have a feeling your issue will be fixed when you upgrade.

**Comment by asamamun** [comment]
> > **IMPORTANT:** If you are posting code or output, please make sure you wrap it with [fenced code blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks/) so that it is readable.
> 
> #### Description
> Tried to use the page from https://simplepie.org/wiki/setup/sample_page to test simplepie. php 8 returns fatal errors about arrays.
> 
> #### Steps to Reproduce
> using the second example, I changed
> 
> ```
> $feed->set_feed_url
> ```
> 
> to
> 
> ```
> $feed->set_feed_url('https://www.usnews.com/rss/news');
> ```
> 
> it gets the title and description perfectly, then I get:
> 
> **_Fatal error: Uncaught TypeError: implode(): Argument #2 ($array) must be of type ?array, string given in /home/a/public_html/simplepie-1.5/library/SimplePie/Parse/Date.php:544_** Stack trace: #0 /home/a/public_html/simplepie-1.5/library/SimplePie/Parse/Date.php(544): implode() #1 /home/a/public_html/simplepie-1.5/library/SimplePie/Parse/Date.php(577): SimplePie_Parse_Date->__construct() #2 /home/a/public_html/simplepie-1.5/library/SimplePie/Registry.php(221): SimplePie_Parse_Date::get() #3 /home/a/public_html/simplepie-1.5/library/SimplePie/Item.php(805): SimplePie_Registry->call() #4 /home/a/public_html/simplepie-1.5/library/SimplePie.php(3203): SimplePie_Item->get_date() #5 [internal function]: SimplePie::sort_items() #6 /home/a/public_html/simplepie-1.5/library/SimplePie.php(3115): usort() #7 /home/a/public_html/simplepie-1.5/demo/test1.php(32): SimplePie->get_items() #8 {main} thrown in /home/a/public_html/simplepie-1.5/library/SimplePie/Parse/Date.php on line 544
> 
> #### Expected Behavior
> I assume it would gracely warn and continue regardless of system date or feed date ?!?
> 
> #### Actual Behavior
> Fatal death?
> 
> #### Possible Solutions
> I guess I can try to see the date setup in php... but other then that this seems like a php8 issue since I believe array functions have changed and many warnings are now errors.
> 
> #### Environment
> php 8.08 on Apache/2.4.38 (Debian)

Replace In Parse/Date.php line 544-545:
$this->day_pcre = '(' . implode(array_keys($this->day), '|') . ')';
$this->month_pcre = '(' . implode(array_keys($this->month), '|') . ')';

to this:
$this->day_pcre = '(' . implode('|',array_keys($this->day)) . ')';
$this->month_pcre = '(' . implode('|',array_keys($this->month)) . ')';

**Comment by Art4** [comment]
@mblaney This issue is fixed since 1.5.3 and can be closed.
